src/core/control/jobs/RenderJob.cpp

```diff
@@ -26,6 +26,8 @@ RenderJob::RenderJob(XojPageView* view): view(view) {}
 auto RenderJob::getSource() -> void* { return this->view; }
 
 void RenderJob::rerenderRectangle(Rectangle<double> const& rect) {
+    this->view->repaintArea(rect.x, rect.y, rect.x + rect.width, rect.y + rect.height);
```

Comment:
I think this should go at the end of the function. The repaint should only happen once the buffer is updated.

Reply:
Yeah, you would usually not see any difference. The RenderJob is run asynchronously, so it may be that `rerenderRectangle` finishes its deed before the repainting actually happens (so you see no difference). But the converse could also happen, leading to the repainting happening too soon.